specs/cli/client-add.md

```diff
@@ -0,0 +1,120 @@
+| `--exclude-path \| -e` | No | \*\*/users/\*\* | A glob pattern to exclude paths from generation. Accepts multiple values. Defaults to no value which excludes nothing. |
+| `--language \| -l` | Yes | csharp | The target language for the generated code files or for the information. |
+| `--namespace-name \| -n` | No | Contoso.GraphApp | The namespace of the client class. Defaults to `Microsoft.Graph`. |
+| `--backing-store \| -b` | No | | Defaults to `false` |
+| `--exclude-backward-compatible \| --ebc` | No |  | Whether to exclude the code generated only for backward compatibility reasons or not. Defaults to `false`. |
+| `--structured-media-types \| -m` | No | `application/json` |Any valid media type which will match a request body type or a response type in the OpenAPI description. Default are documented [here](https://learn.microsoft.com/en-us/openapi/kiota/using#--structured-mime-types--m). |
+| `--skip-generation \| --sg` | No | true | When specified, the generation would be skipped. Defaults to false. |
+| `--output \| -o` | No | ./generated/graph/csharp | The output directory or file path for the generated code files. This is relative to the location of `kiota-config.json`. Defaults to `./output`. |
+
+> [!NOTE] 
+> It is not required to use the CLI to add new clients. It is possible to add a new client by adding a new entry in the `clients` section of the `kiota-config.json` file. See the [kiota-config.json schema](../schemas/kiota-config.json) for more information. Using `kiota client generate --client-name myClient` would be required to generate the code files.
+
+## Using `kiota client add`
+
+```bash
+kiota client add --client-name "graphDelegated" --openapi "https://aka.ms/graph/v1.0/openapi.yaml" --include-path "**/users/**" --language csharp --class-name "GraphClient" --namespace-name "Contoso.GraphApp" --backing-store --exclude-backward-compatible --serializer "Contoso.Json.CustomSerializer" --deserializer "Contoso.Json.CustomDeserializer" -structured-mime-types "application/json" --output "./generated/graph/csharp"
+```
+
+_The resulting `kiota-config.json` file will look like this:_
+
+```jsonc
+{
+  "version": "1.0.0",
+  "clients": {
+    "graphDelegated": {
+      "descriptionLocation": "https://aka.ms/graph/v1.0/openapi.yaml",
+      "includePatterns": ["**/users/**"],
+      "excludePatterns": [],
+      "language": "csharp",
+      "outputPath": "./generated/graph/csharp",
+      "clientClassName": "GraphClient",
+      "clientNamespaceName": "Contoso.GraphApp",
+      "structuredMediaTypes": [
+        "application/json"
+      ],
+      "usesBackingStore": true,
+      "includeAdditionalData": true
+    }
+  }
+}
+```
+
+_The resulting `apimanifest.json` file will look like this:_
+
+```jsonc
+{
+  "apiDependencies": {
+    "graphDelegated": {
+      "x-ms-kiotaHash": "9EDF8506CB74FE44...",
+      "x-ms-kiotaVersion": "1.11.0",
```

Comment:
we could move the version in the kiotaHash. Although having the ability to read it as is might be beneficial to people so they can identify which version was used at glance

Reply:
I think being part of the hash is better and also prevents people playing with the value.